src/popup/components/ProgressBar.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <div class="progress-bar">
+    <div
+      class="progress"
+      :style="progressBarStyle"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent } from '@vue/composition-api';
+
+export default defineComponent({
+  props: {
+    progress: { type: Number, required: true },
+  },
+  setup(props) {
+    const progressBarStyle = computed(() => ({
+      width: `${props.progress}%`,
+    }));
+
+    return {
+      progressBarStyle,
+    };
+  },
```

Comment:
To be honest I don't see a reason for creating whole setup block, computed prop and return it. You can just put this line into the template:
```
  :style="{ width: `${progress}%` }"
```